docarray/base_document/mixins/plot.py

```diff
@@ -0,0 +1,199 @@
+    def schema_summary(cls) -> None:
+        """Print a summary of the Documents schema."""
+        import rich
+
+        panel = rich.panel.Panel(
+            cls._get_schema(), title='Document Schema', expand=False, padding=(1, 3)
+        )
+        rich.print(panel)
+
+    def _ipython_display_(self):
+        """Displays the object in IPython as a side effect"""
+        self.summary()
+
+    @classmethod
+    def _get_schema(cls, doc_name: Optional[str] = None) -> Tree:
+        """Get Documents schema as a rich.tree.Tree object."""
+        import re
+
+        from rich.tree import Tree
+
+        from docarray import BaseDocument, DocumentArray
+
+        name = cls.__name__
+        tree = Tree(name) if doc_name is None else Tree(f'{doc_name}: {name}')
+
+        for k, v in cls.__annotations__.items():
+
+            field_type = cls._get_field_type(k)
+
+            t = str(v).replace('[', '\[')
+            t = re.sub('[a-zA-Z_]*[.]', '', t)
+
+            if is_union_type(v) or is_optional_type(v):
+                sub_tree = Tree(f'{k}: {t}')
+                for arg in v.__args__:
+                    if issubclass(arg, BaseDocument):
+                        sub_tree.add(arg._get_schema())
+                    elif issubclass(arg, DocumentArray):
+                        sub_tree.add(arg.document_type._get_schema())
+                tree.add(sub_tree)
+            elif issubclass(field_type, BaseDocument):
+                tree.add(field_type._get_schema(doc_name=k))
+            elif issubclass(field_type, DocumentArray):
+                field_cls = v.__name__.replace('[', '\[')
+                sub_tree = Tree(f'{k}: {field_cls}')
+                sub_tree.add(field_type.document_type._get_schema())
+                tree.add(sub_tree)
+            else:
+                tree.add(f'{k}: {t}')
+        return tree
+
+    def __rich_console__(self, console, options):
+        kls = self.__class__.__name__
+        id_abbrv = getattr(self, 'id')[:7]
+        yield f":page_facing_up: [b]{kls}" f"[/b]: [cyan]{id_abbrv} ...[cyan]"
+
+        import torch
+        from rich import box, text
+        from rich.table import Table
+
+        import docarray
+
+        table = Table('Attribute', 'Value', width=80, box=box.ROUNDED, highlight=True)
+
+        for k, v in self.__dict__.items():
+            col_1 = f'{k}: {v.__class__.__name__}'
+            if (
+                isinstance(v, (ID, docarray.DocumentArray, docarray.BaseDocument))
+                or k.startswith('_')
+                or v is None
+            ):
+                continue
+            elif isinstance(v, str):
+                col_2 = str(v)[:50]
+                if len(v) > 50:
+                    col_2 += f' ... (length: {len(v)})'
+                table.add_row(col_1, text.Text(col_2))
+            elif isinstance(v, (np.ndarray, torch.Tensor)):
+                if isinstance(v, torch.Tensor):
+                    v = v.detach().cpu().numpy()
```

Comment:
why do we need to do this? Detach can makes sense but to cpu and to numpy I am not sure to understand